plugins/BEdita/Core/src/Command/AsyncJobsCleanCommand.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2024 Atlas Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Command;
+
+use Cake\Command\Command;
+use Cake\Console\Arguments;
+use Cake\Console\ConsoleIo;
+use Cake\I18n\FrozenDate;
+use Cake\ORM\Locator\LocatorAwareTrait;
+
+/**
+ * AsyncJobsClean command.
+ */
+class AsyncJobsCleanCommand extends Command
+{
+    use LocatorAwareTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(Arguments $args, ConsoleIo $io)
+    {
+        $io->info('Cleaning async jobs older than 1 month');
+        $deleted = $this->fetchTable('AsyncJobs')->deleteAll(['created <' => new FrozenDate('-1 month')]);
```

Comment:
Would be nice to provide a `--since` option, which defaults to `-1 month`, to customise the amount of jobs to retain.

It could also be nice to add a `--service` option to clean only jobs for a specific service: one might want to set custom retention policies on a per-service basis (1 month for confirmation emails, 6 months for password recovery attempts, for security purpose). If omitted, jobs for any service are pruned.

Reply:
good suggestions. I'm on it ;)